programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -132,6 +132,110 @@ pub enum SyscallError {
 
 type Error = Box<dyn std::error::Error>;
 
+pub trait HasherImpl {
+    const HASH_BYTES: usize;
+    const NAME: &'static str;
+    type Output: AsRef<[u8]>;
+
+    fn create_hasher() -> Self;
+    fn hash(&mut self, val: &[u8]);
+    fn result(self) -> Self::Output;
+    fn get_base_cost(compute_budget: &ComputeBudget) -> u64;
+    fn get_byte_cost(compute_budget: &ComputeBudget) -> u64;
+    fn get_max_slices(compute_budget: &ComputeBudget) -> u64;
+}
+
+pub struct Sha256Hasher(Hasher);
+pub struct Blake3Hasher(blake3::Hasher);
+pub struct Keccak256Hasher(keccak::Hasher);
+
+impl HasherImpl for Sha256Hasher {
+    const HASH_BYTES: usize = HASH_BYTES;
+    const NAME: &'static str = "Sha256";
+    type Output = [u8; Self::HASH_BYTES];
```

Comment:
we should be able to use each module's `Hash` type here instead of a raw byte array. then we don't need `HASH_BYTES` at all, instead we can replace the usage when calling `translate_slice_mut()` with `std::mem::<H::Output>::size_of()` and call `as_ref()` on it in the `copy_from_slice()` usage

Reply:
done!